src/EditDialog.cpp

```diff
@@ -194,70 +297,79 @@ void EditDialog::setNull()
     ui->editorText->clear();
     ui->editorImage->clear();
     hexEdit->setData(QByteArray());
+    dataType = Null;
 
     // Update the cell data info in the bottom left of the Edit Cell
-    int dataType = checkDataType();
-    updateCellInfo(dataType);
+    updateCellInfo(hexEdit->data());
 
     ui->editorText->setFocus();
 }
 
 void EditDialog::accept()
 {
-    // Don't update if the data hasn't changed
-    // To differentiate NULL and empty byte arrays, we also compare the NULL flag
-    if(hexEdit->data() != oldData || hexEdit->data().isNull() != oldData.isNull())
-    {
-        const QString dataType = ui->comboMode->currentText();
-        bool isBlob = dataType == tr("Binary");
-        emit updateRecordText(curRow, curCol, isBlob, hexEdit->data());
+    // Check if the current cell data is different from the original cell data
+    if (dataSource == TextBuffer) {
+        QString newData = ui->editorText->toPlainText();
+        if ((newData != oldData) || (newData.isNull() != oldData.isNull())) {
```

Comment:
I don't have strong feelings about this either way.  If you have a preference, I'm happy to use it. :smile: